Blockchain/Sources/Blockchain/RuntimeProtocols/RuntimeEvents.swift

```diff
@@ -1,6 +1,41 @@
 import Utils
 
 public enum RuntimeEvents {
+    public struct StateRequestReceived: Event {
+        public var headerHash: Data32
+        public var startKey: Data31
+        public var endKey: Data31
+        public var maxSize: UInt32
+
+        public init(headerHash: Data32, startKey: Data31, endKey: Data31, maxSize: UInt32) {
+            self.headerHash = headerHash
+            self.startKey = startKey
+            self.endKey = endKey
+            self.maxSize = maxSize
+        }
+    }
+
+    public struct StateRequestReceivedResponse: Event {
+        public var headerHash: Data32
+        public var boundaryNodes: [BoundaryNode]
+        public var keyValuePairs: [(key: Data31, value: Data)]
+        public let result: Result<(headerHash: Data32, boundaryNodes: [BoundaryNode], keyValuePairs: [(key: Data31, value: Data)]), Error>
```

Comment:
```suggestion
        public var id: UniqueId
        public let result: Result<(headerHash: Data32, boundaryNodes: [BoundaryNode], keyValuePairs: [(key: Data31, value: Data)]), Error>
```

we need an ID field to know which request is this response for. 
otherwise the response should contain all the request parameters and the id will be all the parameters.

also in the case no response due to error (e.g. no block), there won't be any boundaryNodes etc so they should only exists in the Result type

Reply:
changed